3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,161 @@
+package enrich
+package common
+package enrichments
+package registry
+
+// Java
+import java.net.UnknownHostException
+
+// Maven Artifact
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+
+// Scala
+import scala.util.control.NonFatal
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// json4s
+import org.json4s.JValue
+
+// Iglu
+import iglu.client.{
+  SchemaKey,
+  SchemaCriterion
+}
+import iglu.client.validation.ProcessingMessageMethods._
+
+// Joda-Time
+import org.joda.time.DateTime
+
+// Scala-Forex
+import com.snowplowanalytics.forex.oerclient.{
+  OerClientConfig,
+  AccountType,
+  DeveloperAccount,
+  EnterpriseAccount,
+  UnlimitedAccount,
+  OerResponseError
+}
+import com.snowplowanalytics.forex.{
+  Forex,
+  ForexConfig
+}
+
+// This project
+import common.utils.ConversionUtils
+import utils.ScalazJson4sUtils
+
+/**
+ * Companion object. Lets us create an CurrencyConversionEnrichment
+ * instance from a JValue.
+ */
+object CurrencyConversionEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchema = SchemaCriterion("com.snowplowanalytics.snowplow", "currency_conversion_config", "jsonschema", 1, 0)
+
+  // Creates a CurrencyConversionEnrichment instance from a JValue
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        accountType   <- (ScalazJson4sUtils.extract[String](config, "parameters", "accountType") match {
+          case Success("DEVELOPER")  => DeveloperAccount.success
+          case Success("ENTERPRISE") => EnterpriseAccount.success
+          case Success("UNLIMITED")  => UnlimitedAccount.success
+
+          // Should never happen (prevented by schema validation)
+          case Success(s) => "accountType [%s] is not one of DEVELOPER, ENTERPRISE, and UNLIMITED".format(s).toProcessingMessage.fail
+          case Failure(f) => Failure(f)
+        })
+        rateAt        <- ScalazJson4sUtils.extract[String](config, "parameters", "rateAt")
+        enrich        =  CurrencyConversionEnrichment(accountType, apiKey, baseCurrency, rateAt)
+      } yield enrich).toValidationNel
+    })
+  }
+}
+
+/**
+ * Configuration for a currency_conversion enrichment
+ *
+ * @param apiKey OER authentication
+ * @param baseCurrency Currency to which to convert
+ * @param rateAt Which exchange rate to use - "EOD_PRIOR" for "end of previous day".
+ */
+case class CurrencyConversionEnrichment(
+  accountType: AccountType,
+  apiKey: String,
+  baseCurrency: String,
+  rateAt: String) extends Enrichment {
+
+  val version = new DefaultArtifactVersion("0.1.0")
+
+  val fx = Forex(ForexConfig(nowishCacheSize = 0, nowishSecs = 0, eodCacheSize = 0), OerClientConfig(apiKey, accountType))
```

Comment:
Ouch no good catch! The defaults are fine: https://github.com/snowplow/scala-forex/#311-forexconfig